client/ayon_flame/plugins/create/create_shot_clip.py

```diff
@@ -1,51 +1,555 @@
+            TextDef(
+                "folder",
+                label="{folder}",
+                tooltip="Name of folder used for root of generated shots.\n"
+                        f"{tokens_help}",
+                default=presets.get("folder", "shots"),
+            ),
+            TextDef(
+                "episode",
+                label="{episode}",
+                tooltip=f"Name of episode.\n{tokens_help}",
+                default=presets.get("episode", "ep01"),
+            ),
+            TextDef(
+                "sequence",
+                label="{sequence}",
+                tooltip=f"Name of sequence of shots.\n{tokens_help}",
+                default=presets.get("sequence", "sq01"),
+            ),
+            TextDef(
+                "track",
+                label="{track}",
+                tooltip=f"Name of timeline track.\n{tokens_help}",
+                default=presets.get("track", "{_track_}"),
+            ),
+            TextDef(
+                "shot",
+                label="{shot}",
+                tooltip="Name of shot. '#' is converted to padded number."
+                        f"\n{tokens_help}",
+                default=presets.get("shot", "sh###"),
+            ),
+
+            # verticalSync
+            UILabelDef(
+                label=header_label("Vertical Synchronization Of Attributes")
+            ),
+            BoolDef(
+                "vSyncOn",
+                label="Enable Vertical Sync",
+                tooltip="Switch on if you want clips above "
+                        "each other to share its attributes",
+                default=presets.get("vSyncOn", True),
+            ),
+            EnumDef(
+                "vSyncTrack",
+                label="Hero track",
+                tooltip="Select driving track name which should "
+                        "be mastering all others",
+                items=gui_tracks or ["<nothing to select>"],
+            ),
+
+            # publishSettings
+            UILabelDef(
+                label=header_label("Publish Settings")
+            ),
+            EnumDef(
+                "clipVariant",
+                label="Product Variant",
+                tooltip="Chose variant which will be then used for "
+                        "product name, if <track_name> "
+                        "is selected, name of track layer will be used",
+                items=['<track_name>', 'main', 'bg', 'fg', 'bg', 'animatic'],
+            ),
+            EnumDef(
+                "productType",
+                label="Product Type",
+                tooltip="How the product will be used",
+                items=['plate', 'take'],
+            ),
+            EnumDef(
+                "reviewTrack",
+                label="Use Review Track",
+                tooltip="Generate preview videos on fly, if "
+                        "'< none >' is defined nothing will be generated.",
+                items=['< none >'] + gui_tracks,
+            ),
```

Comment:
To make it aligned with Hiero here is also another difference https://github.com/ynput/ayon-hiero/blob/develop/client/ayon_hiero/plugins/create/create_shot_clip.py#L485-L489

Notice that the attribute is called reviewableSource for a reason. Please take a look at how the flow of the attribute is designed. It is then distributed to the publishing context and used here to determine whether we use review track or clip media for reviewable transcoding downstream. https://github.com/ynput/ayon-hiero/blob/develop/client/ayon_hiero/plugins/publish/collect_plates.py#L34-L47

This of course means that there will be lot of changes and perhaps it would be good to separate them into new PR which would be based on this PR and targeted back to it.

Reply:
Thanks, as suggested I'll look into this as a separate PR.